src/Wrapper/GkmsClientFactory.php

```diff
@@ -4,27 +4,62 @@
 
 namespace Keboola\ObjectEncryptor\Wrapper;
 
+use Google\Auth\HttpHandler\HttpHandlerFactory;
 use Google\Cloud\Kms\V1\KeyManagementServiceClient;
+use GuzzleHttp\BodySummarizer;
+use GuzzleHttp\Client;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Middleware;
 use Keboola\ObjectEncryptor\EncryptorOptions;
 use Keboola\ObjectEncryptor\Exception\ApplicationException;
 use Throwable;
 
 class GkmsClientFactory
 {
+    public const TIMEOUT = 120;
+    public const CONNECT_TIMEOUT = 10;
+
     public function createClient(EncryptorOptions $encryptorOptions): KeyManagementServiceClient
     {
         /* It seems that KeyManagementServiceClient client does not accept retrySettings configuration
             (as some other GCP clients do), therefore we disable retries completely and rely on application level
             retries in GenericGKMWrapper. */
         try {
+            // Create the handler with passed http client with timeout settings
+            $handler = HttpHandlerFactory::build($this->createHttpClient());
+
             // GKM client checks for authorization when created, authorization is cached in memory
             return new KeyManagementServiceClient(
                 [
                     'disableRetries' => true,
+                    'transportConfig' => [
+                        'rest' => [
+                            'httpHandler' => [$handler, 'async'],
+                        ],
+                    ],
```

Comment:
Jeste koukam na streva toho Google clienta a mozna by mohlo misto custom `httpHandler` nastavit jen `restOptions`, co by mela byt primo konfigurace pro Guzzle, jestli mi neco neuslo (jeste to protahnu xdebugem pro overeni)

Reply:
Jo, tak na messengeru mi to zafungovalo. Prochazi to timhle https://github.com/googleapis/google-cloud-php/blob/main/Core/src/RequestWrapper.php#L128, takze jsem misto custom `httpHandler` nastavil `restOptions`, co ze pak preda 1:1 jako options do Guzzle.

@MiroCillik @odinuv muzete mi pls potvrdit, ze tak je je timeout nastaveny tady, tak to chceme vsude? Tzn. connect 10 vterin (fakt?) a cely request 120 vterin?

Reply:
jo, tak to máme všude https://github.com/keboola/object-encryptor/blob/ef25a039c625f89fe7a18edba11c9438cba4aefe/src/Wrapper/KmsClientFactory.php#L15

Reply:
mělo by to failnout relativně rychle a radši se retrynout

Reply:
@pepamartinec jeste se podivam jak to mas v tom messengeru a jestli by to tady neslo jenom pres ty `restOptions` ale nevypadalo to tak.

Reply:
Ne, musi to byt takhle